pilot/pkg/xds/ep_filters.go

```diff
@@ -44,10 +44,14 @@ func (b *EndpointBuilder) EndpointsByNetworkFilter(endpoints []*LocLbEndpointsAn
 	// remote gateways (if any)
 	filtered := make([]*LocLbEndpointsAndOptions, 0)
 
-	// Scale all weights by the maximum number of gateways that might appear for a network.
+	// Scale all weights by the lcm of gateways per network and gateways per cluster.
 	// This will allow us to more easily spread traffic to the endpoint across multiple
 	// network gateways, increasing reliability of the endpoint.
-	scaleFactor := b.push.NetworkManager().GetMaxGatewaysPerNetwork()
+	divisor := uint32(len(b.push.NetworkManager().GatewaysForNetwork(b.network)))
+	if divisor == 0 {
+		divisor = 1
+	}
+	scaleFactor := b.push.NetworkManager().GetLCM() / divisor
```

Comment:
Can we just condense these 4 lines into `NetworkManager().GetScaleFactor(b.network)`

Reply:
Good, can do that

Reply:
done